### ISSUE TITLE
Confluence HTML output has a wrong path for theme.scss on Windows

### ISSUE THREAD

**Issue by m-legrand** [quarto-dev/quarto-cli#4510]
### Bug description

```
$ quarto --version
1.3.217
```

I am on windows, and when trying to render a document with the following frontmatter

```yaml
title: "Example"
format:
    confluence-html
```

I'm getting the following error:

```txt
ERROR: The filename, directory name, or volume label syntax is incorrect. (os error 123), stat 'C:\Users\JDoe\AppData\Local\Programs\Quarto\share\formats\html\bootstrap\themes\C:\Users\JDoe\AppData\Local\Programs\Quarto\share\extensions\quarto\confluence\theme.scss.scss'
```

If I'm decomposing the path in Python code, this looks like

```python
root = "/c/Users/JDoe/AppData/Local/Programs/Quarto/share"
path1 = f"{root}/extensions/quarto/confluence/theme.scss"
path2 = f"{root}/formats/html/bootstrap/themes/{path1}.scss"
```

This hints at a bug whereby `path1` is used somewhere instead of [`path1.stem`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem).
But this is as far as I could go in trying to pinpoint the issue.

---

Github complains about `*.qmd` attachements so I'm giving the source code of a minimal problematic example below:

```markdown
---
title: "Example"
format:
    confluence-html
---

Hello! This is an example.
```

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by allenmanning** [comment]
Thanks for reporting.  I'm looking to reproduce it and will post my findings here.

**Comment by allenmanning** [comment]
On windows I ran the following test.

Quarto Document
````
---
title: "Bug - Confluence HTML output has a wrong path for theme.scss on Windows"
format:
  confluence-html
---

Hello! This is an example.
````

````
quarto render bug-4510.qmd

PS C:\dev\quarto-confluence-test\bugs> quarto render .\bug-4510.qmd
pandoc
  to: html
  output-file: bug-4510.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  theme:
    - default
    - ../../quarto-cli/src/resources/extensions/quarto/confluence/theme.scss
  title: Bug - Confluence HTML output has a wrong path for theme.scss on Windows

Output created: bug-4510.html
````

I wasn't able to reproduce as theme was mapped to `../../quarto-cli/src/resources/extensions/quarto/confluence/theme.scss`

Can you please paste your render output so that I may compare with what I'm seeing.

Thank you.

**Comment by allenmanning** [comment]
I've also tested the preview itself on windows in case the theme wasn't being loaded

`quarto preview bug-4510.qmd`

The preview was opened correctly in a new browser window without any errors being reported in the console.

I'm not sure of the differences between our setups.

**Comment by m-legrand** [comment]
Thanks a lot for picking this up! I appreciate it.

I reproduced the bug from scratch using the same naming conventions and inputs as you in order to help comparison. The error message above is the only output I'm getting, both for `render` and `preview`:

![image](https://user-images.githubusercontent.com/9072763/221158107-621c40d9-d9e4-401e-8a1e-2e6d9296cfd0.png)

**Comment by m-legrand** [comment]
Oh, here is something interesting. I tried doing exactly the same, but this time setting the project folder on my `C:` drive to be even closer to your setup. And... it worked:

![image](https://user-images.githubusercontent.com/9072763/221159166-c7442612-e46e-492b-9f8e-e8cabdf18ae4.png)

I'm guessing this should help pinpointing the issue? Because of idiosyncratic elements, I do need to keep this code on my `D:` drive.

**Comment by allenmanning** [comment]
Thanks for the testing and providing more info.  I've got more to work with now.

**Comment by cderv** [comment]
@allenmanning I can reproduce - here is the stack trace of where the error comes from 
````
    at Object.lstatSync (deno:runtime/js/30_fs.js:305:9)
    at existsSync (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/vendor/deno.land/std@0.166.0/fs/exists.ts:75:10)
    at file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html-scss.ts:183:9
    at Array.flatMap (<anonymous>)
    at layerTheme (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html-scss.ts:179:25)
    at resolveThemeLayer (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html-scss.ts:366:29)
    at resolveBootstrapScss (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html-scss.ts:140:53)
    at boostrapExtras (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html-bootstrap.ts:191:23)
    at themeFormatExtras (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html.ts:894:12)
    at Object.formatExtras (file:///C:/Users/chris/Documents/DEV_R/quarto-cli/src/format/html/format-html.ts:136:11)
````

Here we are joining the theme dir with the theme file 
https://github.com/quarto-dev/quarto-cli/blob/cc03e366904182d2c5f003d69f9f8a59f2d40eb3/src/format/html/format-html-scss.ts#L181

I believe this is supposed to handle theme file without extension, isn't it ? 

In the example here, `theme` is already a full path
````
> theme
'C:/Users/chris/Documents/DEV_R/quarto-cli/src/resources/extensions/quarto/confluence/theme.scss'
````

I am sure we are resolving to this path, when resolving paths for the extensions. We are resolving based on input directory which is here in another drive, which will lead to absolute file instead of `../..` or similar if it was on same drive. 
https://github.com/quarto-dev/quarto-cli/blob/cc03e366904182d2c5f003d69f9f8a59f2d40eb3/src/extension/extension.ts#L371-L386

I don't think `<some path>/../../<end of path>` is better though. Usually extensions are located in project directory but we maybe need to handle specifically the case of extension built in quarto as they will be in the Quarto data folder. (which could be in another drive).

But a quick and scope fixed is to not joined when `theme` is already a full path with extension. 

Hope it helps

**Comment by kjayhan** [comment]
I have a similar problem.

I am updating my syllabus. There was a hyperlink that wasn't working.

I changed it. But somehow, when I render the document, the following happens:

 The filename, directory name, or volume label syntax is incorrect. (os error 123), stat 'D:\Onedrive\R\projects\kjayhan.github.io\syllabi\(https:\www.nyu.edu\about\policies-guidelines-compliance\policies-and-guidelines\academic-integrity-for-students-at-nyu.html)'

It seems that somehow the link got mixed up into a syntax, but how? And how to solve this? Any ideas?

**Comment by allenmanning** [comment]
@kjayhan Are you publishing this to Confluence or another target?  It seems from you message that you are only seeing this when you do a render.  This seems to be a bug specific to multiple drives on windows.  I would suggest that we break this out to be a separate issue.  @cderv FYI

**Comment by kjayhan** [comment]
> @kjayhan Are you publishing this to Confluence or another target? It seems from you message that you are only seeing this when you do a render. This seems to be a bug specific to multiple drives on windows. I would suggest that we break this out to be a separate issue. @cderv FYI

Yes, not Confluence. I have the problem only when I render (I couldn't do publishing yet which is [another problem](https://github.com/quarto-dev/quarto-cli/discussions/4589#discussion-4916767)).

**Comment by cderv** [comment]
@allenmanning @kjayhan yes this is another issue than this one. I'll open a new one from your comment @kjayhan but related to this one as similar problem at the end: 

> This seems to be a bug specific to multiple drives on windows.

I think we'll have a few places in our code where this will be an issue because each time will resolve to relative path, we'll get a absolute path because other drive and we expect a relative one only. 

We'll look into this. Thanks @kjayhan

**Comment by cderv** [comment]
@allenmanning I discussed with @dragonstyle and he will have a look at this issue may be generic for any bundled extensions not just confluence.

**Comment by dragonstyle** [comment]
I believe that this should be fixed here- rather than fixing a more general issue with built in extensions (converting their paths globally to absolute paths), I think for the time being it make sense to make our theme resolution play more nicely with absolute paths. 

https://github.com/quarto-dev/quarto-cli/commit/b9ff0cbc173fa9727ddddbb9a70aaf4513d95dac

The fix should be available in our next pre-release build!

**Comment by cderv** [comment]
Seems good ! 

I find that we have a few fixes that we do now as scoped but that could have maybe broader approach. I am wondering if we should keep track of the broader idea improvment or just leave it as is closed, and trust our memory to find this back in the future :)

**Comment by cderv** [comment]
@m-legrand fix is included in v1.3.247 or later, so please do try it as it should solve the issue for you. 
https://quarto.org/docs/download/prerelease.html

**Comment by dragonstyle** [comment]
TY!

> I find that we have a few fixes that we do now as scoped but that could have maybe broader approach. I am wondering if we should keep track of the broader idea improvment or just leave it as is closed, and trust our memory to find this back in the future :)

As a rule of thumb myself, I try to leave issues open / make an issue if I'm certain that the issue should be addressed. If I have doubts about it, I find it better to leave the issue closed and trust that it will resurface if it merits more investment. 

In this case I though of two possibilities:

1) Specially handle built in extensions (force them always return absolute paths). I don't think this is really necessary as the current system will return absolute paths in the event that a relative path isn't possible (so basically the same thing).

2) Always make extensions return absolute paths. I just am not sure that this is a good idea across the board, so didn't think it made sense to bit this off.

You have thought a lot about this one too (including detailing the issue so well above!), so would be interested in your thoughts!
